php/PhpParser.g4

```diff
@@ -156,22 +152,18 @@ typeParameterWithDefaultsList
     ;
 
 typeParameterDecl
-    : attributes identifier
+    : attributes? identifier
     ;
 
 typeParameterWithDefaultDecl
-    : attributes identifier Eq (qualifiedStaticTypeRef | primitiveType)
+    : attributes? identifier Eq (qualifiedStaticTypeRef | primitiveType)
     ;
 
 genericDynamicArgs
     : '<:' typeRef (',' typeRef)* ':>'
     ;
 
 attributes
-    : attributesGroup*
-    ;
-
-attributesGroup
     : AttributeStart (identifier ':')? attribute (',' attribute)* ']'
```

Comment:
```php
#[Attr1]
#[Attr2]
function f(){}
```
works, https://3v4l.org/8tEvX

Reply:
Thanks, I fixed it and extend attribute.php test file.